### ISSUE TITLE
Editor crash while using C++ class as tool

### ISSUE THREAD

**Issue by lethiandev** [godotengine/godot#19686]
**Godot version:**
3.0.3.stable.official.e649ec7
No-Mono

**OS/device including version:**
Windows 10 x64

**Issue description:**
I was doing some tests with GDNative C++ and i've found that the editor crashes when i'm using the C++ class as a tool in the scene. The tool class itself does nothing special, no memory allocation etc. It happens every time when I minimize and restore the window, then closing the scene tab does the crash.

I did some research on Godot's source code using msvc debugger, and it seems that the GDNative tries to access the memory of the tool class that were already deallocated, or by accessing DLL's heap that doesn't exist anymore?

Some screenshots from the crash:
![image](https://user-images.githubusercontent.com/5323129/41688771-4fa699ea-74e5-11e8-8b59-7852208368f6.png)
![image](https://user-images.githubusercontent.com/5323129/41687112-73df8134-74de-11e8-95b1-0972d090f64a.png)

To debug, I've used custom build of the 3.0 branch (compiled by msvc), the problem appears on the stable 3.0.3 release as well:
![image](https://user-images.githubusercontent.com/5323129/41687696-94d14e48-74e0-11e8-87ef-713d841a6f28.png)

**Steps to reproduce:**
- Open the scene with the native tool class assigned to the node
- Minimize and restore Godot's window
- Close the scene tab

Sorry but I didn't provide linux version of the libraries, instead you can build them by yourself using CMake.
```bash
cd <project_dir> && mkdir build/linux && cd build/linux
cmake -G Ninja -DGDNATIVE_CPP_PATH=<cpp_bindings_dir> -DCMAKE_LIBRARY_OUTPUT_DIRECTORY=../../lib ../..
cmake --build .
```

**Minimal reproduction project:**
[gdnative-tool-class-and-crash.zip](https://github.com/godotengine/godot/files/2121392/gdnative-tool-class-and-crash.zip)

**Comment by karroffel** [comment]
Try setting `reloadable=false` in the `.gdnlib`

**Comment by lethiandev** [comment]
Thanks @karroffel it did the trick. But i think that there should be some mechanism that prevents the editor from the crash anyway. Maybe by tracking the instanced tool classes that prevents the DLL from unload?

**Comment by reduz** [comment]
crash may be unavoidable but just in case I would either document it or avoid it

**Comment by Chaosus** [comment]
Marked as documentation..

**Comment by akien-mga** [comment]
Fixed by #32580.
